test/marginal/test_selection_variable_scikit_learn_classification.py

```diff
@@ -0,0 +1,342 @@
+    """Tests the method on linear cases with noise and/or correlation"""
+    X, y, important_features, _ = data_generator
+
+    importance = configure_marginal_classication(ClassVI, X, y)
+    # check that importance scores are defined for each feature
+    assert importance.shape == (X.shape[1],)
+    # check that important features have the highest importance scores
+    assert np.all([int(i) in important_features for i in np.argsort(importance)[-1:]])
+
+
+parameter_bad_detection = [
+    ("HiDim with high correlated features", 150, 200, 1, 1.0, 42, 1.0, 5.0, 0.0),
+    ("HiDim multivaribale", 150, 200, 10, 0.0, 42, 1.0, np.inf, 0.0),
+    ("HiDim multivaribale noise", 150, 200, 10, 0.0, 42, 1.0, 10.0, 0.0),
+]
+
+
+@pytest.mark.parametrize(
+    "n_samples, n_features, support_size, rho, seed, value, signal_noise_ratio, rho_serial",
+    zip(*(list(zip(*parameter_bad_detection))[1:])),
+    ids=list(zip(*parameter_bad_detection))[0],
+)
+@pytest.mark.parametrize(
+    "ClassVI",
+    [ANOVA, MutualInformationClassification],
+    ids=["ANOVA", "MutualInformation"],
+)
+def test_linear_data_fail(data_generator, ClassVI):
+    """Tests the faillure of the method on linear cases with correlation
+    or multiple variable of importance"""
+    X, y, important_features, _ = data_generator
+    size_support = np.sum(important_features != 0)
+
+    importance = configure_marginal_classication(ClassVI, X, y)
+    # check that importance scores are defined for each feature
+    assert importance.shape == (X.shape[1],)
+    # check that important features have the highest importance scores
+    assert np.any(
+        [
+            int(i) not in important_features
+            for i in np.argsort(importance)[-size_support:]
+        ]
+    )
+
+
+################################################################################
+# Specific test for ANOVA
+parameter_exact_ANOVA = [
+    ("HiDim with high level noise", 150, 200, 1, 0.2, 42, 1.0, 0.5, 0.0),
+]
+
+
+@pytest.mark.parametrize(
+    "n_samples, n_features, support_size, rho, seed, value, signal_noise_ratio, rho_serial",
+    zip(*(list(zip(*parameter_exact_ANOVA))[1:])),
+    ids=list(zip(*parameter_exact_ANOVA))[0],
+)
+def test_ANOVA_exact(data_generator):
+    """Tests the method on high noise"""
+    X, y, important_features, not_important_features = data_generator
+
+    importance = configure_marginal_classication(ANOVA, X, y)
+    # check that importance scores are defined for each feature
+    assert importance.shape == (X.shape[1],)
+    # check that important features have the highest importance scores
+    assert np.all([int(i) in important_features for i in np.argsort(importance)[-1:]])
+    # Check that important features have higher mean importance scores
+    assert (
+        importance[important_features].mean()
+        > importance[not_important_features][
+            np.where(importance[not_important_features] != 0)
+        ].mean()
```

Comment:
```suggestion
        importance[important_features].mean()
        > importance[not_important_features].mean()
```
Is there a specific reason to exclude features with importance=0?

Reply:
That's not a bias; it is expected that the mutual information tends to zero under the null hypothesis (which is the case for for non-important )